src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        Shape[] shapes = new Shape[6];
+        ShapeSupplier generatedShape = new ShapeSupplier();
```

Comment:
The `ShapeSupplier` instance is correctly created outside the loop, which is a good practice as per the checklist requirements .